bin/compile.py

```diff
@@ -0,0 +1,337 @@
+#!/usr/bin/env python3
+import codecs
```

Comment:
Please follow PEP8 for imports: https://www.python.org/dev/peps/pep-0008/#imports

Specifically, the following:

>Imports should be grouped in the following order:
> 1. Standard library imports.
> 2. Related third party imports.
> 3. Local application/library specific imports.
>
>You should put a blank line between each group of imports.